### PR TITLE
Add Nexus OSS Repository Manager to TrueNAS community plugins

### PR DESCRIPTION
**Pull request by spolti**
Thank you for your submission to the FreeNAS community plugins repository!

Please ensure the following guidelines are met when submitting a new Plugin.

1. Add the entry to INDEX in alphabetical order
2. Includes a new icon in the ./icon/ directory
3. Ensure the submitted icon file is a valid PNG that is 128x128 in size
4. Add new plugin cirrus task

### CODE REVIEW

File: .cirrus.yml [hunk 1]
```diff
@@ -311,6 +311,15 @@ netdata_task:
   env:
     PLUGIN_FILE: "netdata.json"
 
+nexus_task:
+  <<: *INSTALL_PLUGIN
+  only_if: "changesInclude('nexus-oss.json', '.cirrus/install_script.sh')"
+  matrix:
+    - freebsd_instance:
+        image_family: freebsd-12-1
```
Comment (tprelog) [inline .cirrus.yml:319]: Probably want to use `12-2` here
```suggestion
        image_family: freebsd-12-2
```

**Review by sonicaj — COMMENTED** [review]
Let's please have a `master` branch or `12.2-RELEASE` branch in the plugin repository as `main` right now isn't used by iocage. We will give preference to `12.2-RELEASE` branch and if that does not exist, fall back to `master`. We can perhaps add this as a fallback in iocage as well but for now can you please make sure either of those branches exist ? Thank you

File: .cirrus.yml (hunk 1 above)
Comment (spolti) [reply .cirrus.yml]: yeah, nice catch.

File: INDEX [hunk 2]
```diff
@@ -303,6 +303,15 @@
...
 		"official": false
 	},
+    "nexus-oss": {
+        "MANIFEST": "nexus-oss.json",
+        "name": "NexusOSS",
+        "icon": "https://icons.freenas.org/community-icons/nexus-oss.png",
+        "description": "Nexus OSS Repository Manager",
+        "maintainer": "Filippe Spolti <filippespolti@gmail.com>",
+        "official": false,
+        "primary_pkg": "null"
```
[the remark this answers is not in the document]
Comment (spolti) [reply INDEX:313]: ok, updated.

File: nexus-oss.json [hunk 3]
```diff
@@ -0,0 +1,23 @@
+{
+  "name": "nexus-oss",
+  "plugin_schema": "2",
+  "release": "12.2-RELEASE",
+  "artifact": "https://github.com/rebasing-xyz/iocage-nexus-oss.git",
+  "official": false,
+  "properties": {
+    "dhcp": 1
```
[the remark this answers is not in the document]
Comment (spolti) [reply nexus-oss.json:8]: probably yes, didn't try yet.
will give a shot and update this PR, thanks.

[the remark this answers is not in the document]
Comment (spolti) [reply nexus-oss.json:8]: Yeah, it works, updated as well to use nat.

**Comment by spolti** [comment]
Hi, sry, I've missed the last message about the branch, I have created the 12.2-RELEASE branch.
https://github.com/rebasing-xyz/iocage-nexus-oss/tree/12.2-RELEASE

Do I need to make it default or can it be configured on the test side?

**Comment by fulder** [comment]
> Do I need to make it default or can it be configured on the test side?

The tests are (just like `iocage`)  looking for the release branch (specified in the `.cirrus.yml` file) and if it's not found falling back to the `master` instead. So it should work just without any other changes needed when it comes to the plugin repo.

Can you try to add a dummy update to the plugin JSON manifest (e.g. extra space and then revert it again)? This should re-run the test for this plugin again.

**Review by sonicaj — APPROVED** [review]
Thank you for your contribution @spolti.

**Comment by sonicaj** [comment]
@fulder the test can be restarted via cirrus, so that is not required ;)

**Comment by spolti** [comment]
Thanks :)

**Pull request merged**